kernel/src/hil/digest.rs

```diff
@@ -41,6 +68,16 @@ pub trait Digest<'a, const L: usize> {
         data: LeasableBuffer<'static, u8>,
     ) -> Result<usize, (ErrorCode, &'static mut [u8])>;
 
+    /// Clear the keys and any other sensitive data.
+    /// This won't clear the buffers provided to this API, that is up to the
+    /// user to clear.
+    fn clear_data(&self);
+}
+
+/// Computes a digest (cryptographic hash) over data
+///
+/// 'L' is the length of the 'u8' array to store the digest output.
+pub trait DigestHash<'a, const L: usize> {
     /// Request the hardware block to generate a Digest and stores the returned
```

Comment:
Shouldn't this have a `set_client`?

Note that for `UART`, to make things a little simpler, it disambiguates the different clients through method names. So `set_digest_client`.